Content.Server/Decals/DecalSystem.cs

```diff
@@ -0,0 +1,277 @@
+                throw new ArgumentOutOfRangeException($"Tried to set decal id to invalid prototypeid: {id}");
+
+            var chunkCollection = ChunkCollection(gridId);
+            var decal = chunkCollection[indices][uid];
+            chunkCollection[indices][uid] = decal.WithId(id);
+            DirtyChunk(gridId, indices);
+            return true;
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var playerSession in _playerManager.GetAllPlayers())
+            {
+                var chunks = GetChunksForSession(playerSession);
+                var updatedChunks = new Dictionary<GridId, HashSet<Vector2i>>();
+                foreach (var (gridId, gridChunks) in chunks)
+                {
+                    var newChunks = new HashSet<Vector2i>(gridChunks);
+                    if (_previousSentChunks[playerSession].TryGetValue(gridId, out var previousChunks))
+                    {
+                        newChunks.ExceptWith(previousChunks);
+                    }
+
+                    if (_dirtyChunks.TryGetValue(gridId, out var dirtyChunks))
+                    {
+                        gridChunks.IntersectWith(dirtyChunks);
+                        newChunks.UnionWith(gridChunks);
+                    }
+
+                    if (newChunks.Count == 0)
+                        continue;
+
+                    updatedChunks[gridId] = newChunks;
+                }
+
+                if(updatedChunks.Count == 0)
+                    continue;
+
+                _previousSentChunks[playerSession] = chunks;
+
+                //send all gridChunks to client
+                SendChunkUpdates(playerSession, updatedChunks);
+            }
+
+            _dirtyChunks.Clear();
+        }
+
+        private void SendChunkUpdates(IPlayerSession session, Dictionary<GridId, HashSet<Vector2i>> updatedChunks)
+        {
+            var updatedDecals = new Dictionary<GridId, Dictionary<Vector2i, Dictionary<uint, Decal>>>();
+            foreach (var (gridId, chunks) in updatedChunks)
+            {
+                var gridChunks = new Dictionary<Vector2i, Dictionary<uint, Decal>>();
+                foreach (var indices in chunks)
+                {
+                    gridChunks.Add(indices,
+                        ChunkCollection(gridId).TryGetValue(indices, out var chunk)
+                            ? chunk
+                            : new Dictionary<uint, Decal>());
+                }
+                updatedDecals[gridId] = gridChunks;
+            }
+
+            RaiseNetworkEvent(new DecalChunkUpdateEvent{Data = updatedDecals}, Filter.SinglePlayer(session));
+        }
+
+        private HashSet<EntityUid> GetSessionViewers(IPlayerSession session)
+        {
+            var viewers = new HashSet<EntityUid>();
+            if (session.Status != SessionStatus.InGame || session.AttachedEntityUid is null)
+                return viewers;
```

Comment:
Elsewhere you check for Connected but this checks for InGame.

Reply:
ye as i said above, only need to send it to them when they are actually ingame, so i should also only create the index for players that are ingame